app/services/process_logs_service.rb

```diff
@@ -16,22 +16,159 @@ def call
       Rails.logger.error(e.message)
     end
   end
+class ProcessLogsService < BaseLogsService
+  def initialize(log_ids)
+    super()
+
+    @log_ids = log_ids
+  end
+
+  def call
+    Rails.logger.info("Received log_ids: #{@log_ids.inspect}")
+
+    begin
+      Travis::Lock.exclusive('process_logs', lock_options) do
+        update_log_entries
+      end
+    rescue => e
+      Rails.logger.error(e.message)
+    end
+  end
```

Comment:
@Jullianos @AndriiMysko what this code is doing here?

Reply:
Based on the requirements once the process job gets picked up from the queue it should fetch the logs by the provided log_ids, update their status to :started and create new ScanTrackerEntry rows, and finally it should write the log content into a local directory/file from where it will be picked up and censored.

Reply:
@Jullianos pls review the whole file. You have duplicated code after rebase here.

Reply:
Yes, stupid mistake, thanks for noticing.